lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -118,14 +119,14 @@ def _proc_upload_reads_params(self, params):
                              'with a local forward reads file')
         if not shock and revfile:
             revid = os.path.abspath(os.path.expanduser(revfile))
-        interleaved = 1 if params.get('interleaved') else 0
+        interleaved = 0
+        if params.get('interleaved') or revid:
+            interleaved = 1
         kbtype = 'KBaseFile.SingleEndLibrary'
         single_end = True
         if interleaved or revid:
```

Comment:
Not identical.  There is a rev_id check as well. I find the way I have the code more readable.